ui/src/actions/search.js

```diff
@@ -1,149 +1,179 @@
-import { push } from 'connected-react-router';
 import { stringify } from 'qs';
+import { push } from 'connected-react-router';
 
 import {
   SEARCH_REQUEST,
   SEARCH_ERROR,
   SEARCH_SUCCESS,
-  CHANGE_SEARCH_SCOPE,
   SEARCH_AGGREGATIONS_REQUEST,
   SEARCH_AGGREGATIONS_SUCCESS,
   SEARCH_AGGREGATIONS_ERROR,
+  SEARCH_QUERY_UPDATE,
   NEW_SEARCH_REQUEST,
+  CHANGE_SEARCH_BOX_NAMESPACE,
+  SEARCH_BASE_QUERIES_UPDATE,
 } from './actionTypes';
 import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';
 import { httpErrorToActionPayload } from '../common/utils';
+import { FETCH_MODE_NEVER, FETCH_MODE_INITIAL } from '../reducers/search';
 
-export function changeSearchScope(scope) {
-  return {
-    type: CHANGE_SEARCH_SCOPE,
-    payload: scope,
-  };
+function getQueryForCurrentState(namespace, state) {
```

Comment:
This is defined also later but for current state, maybe could be extracted somewhere and we pass the current state in the other case?

Reply:
it's only `getIn(['namespaces', namespace, 'query'])` basically, that's I just didn't wanna create common functions everywhere, and I created for places just to make the main functions that use this, a bit more readable.

But I am considering adding helpers for paths like `['namespaces', namespace, 'query']`